scout/adapter/mongo/case.py

```diff
@@ -883,9 +874,25 @@ def check_existing_data(self, case_obj, existing_case, institute_obj, update, ke
             )
 
         if keep_actions:
+            # collect eventual phenotype-related data already existing for the case
+            self.update_case_phenotypes(old_case=existing_case, new_case=case_obj)
```

Comment:
Wouldn't this be better called over here?
https://github.com/Clinical-Genomics/scout/blob/6a68ef2636622cfc1ef14e117b85f523007f50f3/scout/adapter/mongo/case.py#L1010

Reply:
I wouldn't mix data-sharing and phenotype-related keys if the phenotype-related ones have to be checked against, and eventually be integrated with keys present on the config file. It would be more messy I think.

Reply:
Ah OK, but I think you din't meant, joining the keys of the two functions into a unique function. Then yes, why not, I'll move the call to the function in the finally, as you suggest!